doc/guides/west/build-flash-debug.rst

```diff
@@ -19,6 +22,71 @@ build system commands with the same names directly delegate to West.
    .. contents::
       :local:
 
+.. _west-building:
+
+Building: ``west build``
+************************
+
+.. tip:: Run ``west build -h`` for additional help.
+
+The ``build`` command allows you to build any source tree from any directory
+in your file system, placing the build results in a folder of your choice.
+
+In its simplest form, the command can be run by navigating to the root folder
+(i.e. the folder containing a file:`CMakeLists.txt` file) of the Zephyr
+application of your choice and running::
+
+  west build -b <BOARD>
+
+Where ``<BOARD>`` is the name of the board you want to build for. This is
+exactly the same name you would supply to CMake if you were to invoke it with:
+``cmake -DBOARD=<BOARD>``.
+A build folder named :file:`./build` (default name) will be created and the
+build output will be placed in it.
+
+To specify the build directory, use ``--build-dir`` (or ``-d``)::
+
+  west build -b <BOARD> --build-dir path/to/build/directory
+
+Since the build directory defaults to :file:`./build`, if you do not specify
+a build directory but a folder named `./build` is present that will be used,
+allowing you to incrementally build from outside the file:`build` folder with
+no additional parameters.
+
+.. note::
+  The ``-b <BOARD>`` parameter is only required when there is no CMake cache
+  present at all, usually when you are building from scratch or creating a
+  brand new build directory. After that you can rebuild (even with additional
+  parameters) without having to specify the board again.
+
+To specify the source directory, use ``--source-dir`` (or ``-s``)::
+
+  west build -b <BOARD> --source-dir path/to/source/directory
+
+Additionally you can specify the target to be specified to the build system
+using the ``--target`` (or ``-t``) option::
+
+  west build -t clean
+
+You can list all targets with ``ninja help`` inside the build folder.
+
+Finally, you can also fine-tune the CMake invocation that happens under the
+hood by using the ``--cmake`` (or ``-c``) option to force CMake to be run
```

Comment:
So I think that simply passing `--cmake` does not fine tune the CMake invocation. It just forces CMake to be re-run. Also, some of the descriptive text associated with west build commands got a bit mixed up.

What about this:

----

Finally, you can add additional arguments to the CMake invocation performed by `west build` by supplying additional parameters (after a `--`) to the command. For example, to use the Unix Makefiles CMake generator instead of Ninja (which `west build` uses by default), run:

```
   west build -- -G'Unix Makefiles'
```

As another example, the following command adds the `file.conf` Kconfig fragment to the files which are merged into your final build configuration:

```
   west build -- -DOVERLAY_CONFIG=file.conf
```


Passing additional CMake arguments like this forces `west build` to re-run CMake, even if a build system has already been generated. You can also force a CMake re-run using the `-c` (or `--cmake`) option, as in `west build --cmake`.

----

(I've left the `-b <BOARD>` out of the above to illustrate that it's not always needed, but it could also make sense to put that in each command for consistency.)